core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -728,17 +728,17 @@ public void removeBlocks(Collection<Long> blockIds, boolean delete) throws Unava
   @Override
   public void validateBlocks(Function<Long, Boolean> validator, boolean repair)
       throws UnavailableException {
+    long limit = 100;
     List<Long> invalidBlocks = new ArrayList<>();
     try (CloseableIterator<Block> iter = mBlockMetaStore.getCloseableIterator()) {
-      while (iter.hasNext()) {
+      while (iter.hasNext() && invalidBlocks.size() < limit) {
         long id = iter.next().getId();
         if (!validator.apply(id)) {
           invalidBlocks.add(id);
         }
       }
     }
     if (!invalidBlocks.isEmpty()) {
-      long limit = 100;
       List<Long> loggedBlocks = invalidBlocks.stream().limit(limit).collect(Collectors.toList());
       LOG.warn("Found {} orphan blocks without corresponding file metadata.", invalidBlocks.size());
       if (invalidBlocks.size() > limit) {
```

Comment:
If we put the over limit check to the while condition, this condition cannot be true forever?

Reply:
you are right, this change is added by mistake. It is supposed to be part of another change.
Roll it back